edweiss-app/__test__/components/lectures/remotecontrol/modal.test.tsx

```diff
@@ -0,0 +1,168 @@
+import LangSelectModal from '@/components/lectures/remotecontrol/modal';
+import LectureDisplay from '@/model/lectures/lectureDoc';
+import { BottomSheetModal } from '@gorhom/bottom-sheet';
+import { fireEvent, render } from '@testing-library/react-native';
+import React from 'react';
+
+const mockSetLang = jest.fn();
+const mockOnClose = jest.fn();
+
+const availableLangs: LectureDisplay.AvailableLangs[] = [
+    'english', 'french', 'german', 'spanish', 'italian', 'brazilian', 'arabic', 'chinese', 'vietanames', 'hindi'
+];
+
+
+
+// Mock translation function
+jest.mock('@/config/i18config', () => ({
+    __esModule: true,
+    default: jest.fn((key) => key),
+}));
+
+// Expo router with stack screen to test up buttons and title
+jest.mock('expo-router', () => {
+    const { Text } = require('react-native');
+    return {
+        router: { push: jest.fn() },
+        Stack: {
+            Screen: jest.fn(({ options }) => (
+                <>
+                    <Text testID="title">{options.title}</Text>
+                </>
+            )),
+        },
+    };
+});
+
+// Authentication mock
+jest.mock('@/contexts/auth', () => ({
+    useAuth: jest.fn(),
+}));
+
+// BottomSheet to detect modal appearance
+jest.mock('@gorhom/bottom-sheet', () => ({
+    BottomSheetModal: jest.fn(({ children }) => (
+        <div>{children}</div>
```

Comment:
Isn't it better for testing purposes to use TView instead of div since it will correlate better with the way we do things? Or is there another aspect of div that makes it more valuable in this scenario?

Reply:
So that's a great question that you ask. I don't know if you had to test `React.Component` with`BottomSheetModal`. So the way it works is that it will replace it with the new provided mocked function that takes as input the `children` and `wrap it arround` into a `div` which is `React` style not `React.Native`.
However, this is only to reduce the code overhead, as it has been proven to work independently of the <TView> which in our case also needs to be mocked (with a `div` 😁).
Therefore, this mock includes the minimal amount of information to make it displayable and testable, with the minimal code overhead. 

Hope you understand. Is it more clear this way ?

Reply:
Yes thank you